python/shotgun_data/sgdata.py

```diff
@@ -690,34 +700,60 @@ def run(self):
 
                 elif item_type == ShotgunDataRetriever._THUMB_DOWNLOAD:
                     # download the actual thumbnail. Because of S3, the url
-                    # has most likely expired, so need to re-fetch it via a sg find
+                    # may have expired - in that case fall back, get a fresh url
+                    # from shotgun and try again
                     entity_id = item_to_process["entity_id"]
                     entity_type = item_to_process["entity_type"]
                     field = item_to_process["field"]
-
-                    sg_data = self.__sg.find_one(entity_type, [["id", "is", entity_id]], [field])
```

Comment:
Avoiding this find() lookup happening for every thumbnail drastically speeds up thumbnail loading! Now instead, the find() lookup only happens when the S3 url doesn't work (likely because of expiry).